addons/account/models/account_payment_term.py

```diff
@@ -41,18 +60,45 @@ def _compute_fiscal_country_codes(self):
             allowed_companies = record.company_id or self.env.companies
             record.fiscal_country_codes = ",".join(allowed_companies.mapped('account_fiscal_country_id.code'))
 
+    def _get_discount_amount_currency(self, total_amount, untaxed_amount):
+        self.ensure_one()
+        if self.early_discount:
+            percentage = self.discount_percentage / 100.0
+            if self.early_pay_discount_computation in ('excluded', 'mixed'):
+                self.discount_amount_currency = self.currency_id.round((total_amount - untaxed_amount) * percentage)
```

Comment:
You must not assign `discount_amount_currency` from `_compute_example_preview` because `discount_amount_currency` is not supposed to be computed by this compute